core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/ScanSourceOpDesc.scala

```diff
@@ -65,14 +66,9 @@ abstract class ScanSourceOpDesc extends SourceOperatorDescriptor {
       throw new RuntimeException("no input file name")
     }
 
-    if (AmberConfig.isUserSystemEnabled) {
-      // if user system is defined, a datasetFileDesc will be initialized, which is the handle of reading file from the dataset
-      datasetFile = Some(new DatasetFileDocument(Paths.get(fileName.get)))
-    } else {
-      // otherwise, the fileName will be inputted by user, which is the filePath.
-      filePath = fileName
-    }
-
+    val (resolvedPath, resolvedDatasetFile) = ScanSourceFileResolver.resolve(fileName.get)
+    filePath = resolvedPath
+    datasetFile = resolvedDatasetFile
```

Comment:
You could combine both into filePath as Either

Reply:
can you elaborate this one?